### PR TITLE
Add option for disabling dependency analysis in run phase

### LINKED ISSUE #785: Xcodeproj doesn't know about the following attributes {"alwaysOutOfDate"=>"1"}
I get this error when trying to run `pod install` on my project.

This attribute is enabled by unchecking the "Based on dependency analysis" option in a "Run Script" build phase (see screenshot).

Screen Shot 2020-09-22 at 22 26 54

### PR DESCRIPTION
**Pull request by LemonSpike**
closes #785

### CODE REVIEW

File: lib/xcodeproj/project/object/build_phase.rb [hunk 1]
```diff
@@ -29,6 +29,17 @@ class AbstractBuildPhase < AbstractObject
...
+        
+        # @return [String] whether or not this run script will be forced to 
+        #         run even on incremental builds. Can be either '1', or 
+        #         missing. By default this option is disabled in Xcode.
+        #
+        # @note   This setting is exposed in Xcode in the UI of
+        #         PBXShellScriptBuildPhase as `Based on
+        #         dependency analysis` (selected by default).
+        #
+
```
Comment (dnkoutso) [inline lib/xcodeproj/project/object/build_phase.rb:41]: nit empty line

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@
 * Add support of c++ files (`hpp` and `cpp`) in `PBXFileReference:set_last_known_file_type`.  
   [romanmikhailov](https://github.com/romanmikhailov)
   [#783](https://github.com/CocoaPods/Xcodeproj/issues/783)
+* Add support of dependency analysis option in `PBXShellScriptBuildPhase:always_out_of_date`.
```
Comment (dnkoutso) [inline CHANGELOG.md:10]: nit: add one line here.

Also add two empty spaces at the end of this line for markdown formatting.

File: lib/xcodeproj/project/object/build_phase.rb [hunk 3]
```diff
@@ -29,6 +29,16 @@ class AbstractBuildPhase < AbstractObject
...
+        
+        # @return [String] whether or not this run script will be forced to 
+        #         run even on incremental builds. Can be either '1', or 
+        #         missing. By default this option is disabled in Xcode.
+        #
+        # @note   This setting is exposed in Xcode in the UI of
+        #         PBXShellScriptBuildPhase as `Based on
+        #         dependency analysis` (selected by default).
+        #
+        attribute :always_out_of_date, String
```
Comment (dnkoutso) [inline lib/xcodeproj/project/object/build_phase.rb:41]: shoudl the default value be '0' like in the one above? Or is that already the case?

Comment (LemonSpike) [reply lib/xcodeproj/project/object/build_phase.rb:41]: the default is for the whole option doesn't appear in the pbxproj. Only when you disable it it comes up as having the value 1.

Comment (dnkoutso) [reply lib/xcodeproj/project/object/build_phase.rb:41]: sounds good

**Comment by dnkoutso** [comment]
Need to run `bundle exec rubocop -a` to fix linter issues.

**Comment by dnkoutso** [comment]
Can you also rebase/squash? I can do it this during merge too.

**Review by dnkoutso — APPROVED** [review]

**Comment by LemonSpike** [comment]
Is there a getting started guide for contributing to this repo and installing the gems on Mac? I'm getting build errors with sudo `gem install ffi -v '1.9.17' --source 'https://rubygems.org/'`, so not sure what ruby version to use to get `bundle install` working.

**Comment by LemonSpike** [comment]
@dnkoutso Rebase/squash done alongside rubocop now. :) Hope we can merge this now.

**Review by amorde — APPROVED** [review]

**Pull request merged**